README.md

```diff
@@ -1 +1,133 @@
 # xtractmime
+
+`xtractmime` is a [BSD-licensed](https://opensource.org/licenses/BSD-3-Clause)
+Python 3.6+ implementation of the [MIME Sniffing
+Standard](https://mimesniff.spec.whatwg.org/).
+
+Install from [`PyPI`](https://pypi.python.org/pypi/xtractmime):
+
+```
+pip install xtractmime
+```
+
+---
+
+## Basic usage
+
+Below mentioned are some simple examples of using `xtractmime.extract_mime`:
+
+```python
+>>> from xtractmime import extract_mime
+>>> extract_mime(b'Sample text content')
+b'text/plain'
+>>> extract_mime(b'', content_types=(b'text/html',))
+b'text/html'
+```
+
+Additional functionality to check if a MIME type belongs to a specific MIME type group using 
+methods mentioned in `xtractmime.mimegroups`:
+
+```python
+>>> from xtractmime.mimegroups import is_html_mime_type, is_image_mime_type
+>>> mime_type = b'text/html'
+>>> is_html_mime_type(mime_type)
+True
+>>> is_image_mime_type(mime_type)
+False
+```
+
+---
+
+## API Reference
+
+### function `xtractmime.extract_mime : Optional[bytes]`
+**Parameters:**
+
+* `body: bytes`
+* `content_types: Optional[Tuple[bytes]], default = None`
+* `http_origin: bool, default = True`
+* `no_sniff: bool, default = False`
+* `extra_types: Optional[Tuple[Tuple[bytes, bytes, Optional[Set[bytes]], bytes], ...]], default = None`
+* `supported_types: Set[bytes], default = None`
+
+Return the [MIME type essence](https://mimesniff.spec.whatwg.org/#mime-type-essence) (e.g. `text/html`) matching the input data, or 
+`None` if no match can be found.
+
+The `body` parameter is the byte sequence of which MIME type is to be determined. `xtractmime` only consider first few
+bytes of the `body` and the specific number of bytes read is defined in the `xtractmime.RESOURCE_HEADER_BUFFER_LENGTH` constant.
+
+Optional `content_types` argument is a tuple of mime types suitable for using as a `Content-Type` header.
+For the cases where the format of the content is unrecognizable for `xtractmime`, it will return
+the last mime type in the provided tuple of mime types.
```

Comment:
“Optional” is already indicated in the parameters signature, so it does not add much here. With code formatting and after the parameters signature, we should also be able to refer to parameters by name directly, without mentionign they are arguments.

```suggestion
`content_types` is a tuple of MIME types suitable for using as a `Content-Type` header.
For the cases where the format of the content is unrecognizable for `xtractmime`, it will return
the last mime type in the provided tuple of mime types.
```

Also, we should not make this about the `Content-Type` headers. We should indicate that, for resources retrieved via HTTP, this should be the list of MIME types indicated in the `Content-Type` response header. However, for a file system or other protocols (e.g. FTP) this should contain whatever MIME type those indicated for the resource.

Reply:
> Also, we should not make this about the `Content-Type` headers. We should indicate that, for resources retrieved via HTTP, this should be the list of MIME types indicated in the `Content-Type` response header. However, for a file system or other protocols (e.g. FTP) this should contain whatever MIME type those indicated for the resource.

Right, currently it seems that the argument is particular to `Content-Type` header. Maybe we can change to this
> `content_types` is a tuple of MIME types given in the resource metadata and can be used as a
`Content-Type` header for the responses retrieved via HTTP. For the cases where the format of the content is unrecognizable for `xtractmime`, it will return the last MIME type in the provided tuple of mime types.

Reply:
> […] can be used as a `Content-Type` header for the responses retrieved via HTTP.

This part does not sound right to me. It sounds like the parameter to `xtractmime` can be used to build a response header, but it should sound more like you can use a response header to build a parameter for `xtractmime`.

> For the cases where the format of the content is unrecognizable for `xtractmime`, it will return the last MIME type in the provided tuple of MIME types.

I think we should omit this information altogether, as it is an implementation detail that users should not rely on. For example, if because of this note users pass only the last value (since the other values will not be used at all), if in the future the standard changes to take only the last type that is valid, or that is not known to the standard, then we have an issue.